libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/NavigationCustomEvent.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.os.Parcel
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.Event
+import com.mapbox.android.telemetry.TelemetryUtils
+import com.mapbox.navigation.base.metrics.MetricEvent
+import com.mapbox.navigation.base.metrics.NavigationMetrics
+
+@SuppressLint("ParcelCreator")
+internal class NavigationCustomEvent : Event(), MetricEvent {
+
+    private companion object {
+        private val OPERATING_SYSTEM = "Android - ${Build.VERSION.RELEASE}"
+    }
+
+    var type: String = ""
+    var payload: String? = null
+    val version: String = "2.4"
+    var customEventVersion: String = "1.0.0"
+    val event: String = NavigationMetrics.CUSTOM_EVENT
+
+    val created: String = TelemetryUtils.obtainCurrentDate()
+    val createdMonotime: Int = 123
+    val operatingSystem: String = OPERATING_SYSTEM
+    val device: String? = Build.MODEL
+    val driverMode = "trip"
+    val driverModeId: String = TelemetryUtils.obtainUniversalUniqueIdentifier()
+    val driverModeStartTimestamp: String = TelemetryUtils.obtainCurrentDate()
+    val driverModeStartTimestampMonotime: Int = 123
```

Comment:
What's that?

Reply:
This property doesn't make sense for Drop-In UI. However since it is required by the schema, I have changed it to var so that any other SDK feature using custom event can modify it.